crates/rrg/src/fs.rs

```diff
@@ -81,48 +68,14 @@ pub fn walk_dir<P: AsRef<Path>>(root: P) -> std::io::Result<WalkDir> {
     };
 
     Ok(WalkDir {
+        max_depth: u32::MAX,
         iter,
         pending_iters: vec![],
         #[cfg(target_family = "unix")]
         dev: dev,
     })
 }
 
-/// Returns a shallow iterator over entries within a directory.
-///
-/// This function is very similar to the standard [`read_dir`], except that the
-/// returned iterator always returns entries with valid metadata and a path.
-///
-/// [`read_dir`]: std::fs::read_dir
-///
-/// # Errors
-///
-/// This function will fail if the specified path does not represent a directory
-/// or does not exist. Note that items returned by the iterator can also contain
-/// errors in case there was an issue with a particular directory entry.
-///
-/// # Examples
-///
-/// ```no_run
-/// use std::path::PathBuf;
-///
-/// let paths = rrg::fs::list_dir("/").unwrap()
-///     .filter_map(Result::ok)
-///     .map(|entry| entry.path)
-///     .collect::<Vec<_>>();
-///
-/// assert!(paths.contains(&PathBuf::from("/home")));
-/// assert!(paths.contains(&PathBuf::from("/bin")));
-/// assert!(paths.contains(&PathBuf::from("/tmp")));
-/// ```
-pub fn list_dir<P: AsRef<Path>>(path: P) -> std::io::Result<ListDir> {
-    let iter = std::fs::read_dir(path)?;
-
-    Ok(ListDir {
-        iter: iter,
-    })
-}
-
 /// Iterator over entries in all subdirectories.
```

Comment:
add "until `max_depth`"?

Reply:
`max_depth` is a private field of the struct and docs are for public consumption, it should not leak implementation details.

However, I added a paragraph with reference to the `with_max_depth` method.